python/lsst/drp/tasks/gbdesAstrometricFit.py

```diff
@@ -1407,3 +1477,496 @@ def _compute_model_params(self, wcsf):
+        visits will be put into the same `wcsfit.Field` and fit together.
+
+        Paramaters
+        ----------
+        inputVisitSummaries : `list` of `lsst.afw.table.ExposureCatalog`
+            List of catalogs with per-detector summary information.
+
+        Returns
+        -------
+        fields : `wcsfit.Fields`
+            Object with field information.
+        fieldRegions : `dict` [`int`, `lsst.sphgeom.ConvexPolygon`]
+            Dictionary of regions encompassing each group of input visits,
+            keyed by an arbitrary index.
+        """
+        allDetectorCorners = []
+        mjds = []
+        radecs = []
+        radii = []
+        for visSum in inputVisitSummaries:
+            detectorCorners = [
+                lsst.geom.SpherePoint(ra, dec, lsst.geom.degrees).getVector()
+                for (ra, dec) in zip(visSum["raCorners"].ravel(), visSum["decCorners"].ravel())
+            ]
+            allDetectorCorners.append(detectorCorners)
+
+            # Get center and approximate radius of field of view
+            hull = lsst.sphgeom.ConvexPolygon.convexHull(detectorCorners)
+            boundingCircle = lsst.sphgeom.ConvexPolygon.convexHull(detectorCorners).getBoundingCircle()
+            center = lsst.geom.SpherePoint(boundingCircle.getCenter())
+            ra = center.getRa().asDegrees()
+            dec = center.getDec().asDegrees()
+            radecs.append([ra, dec])
+            radius = boundingCircle.getOpeningAngle()
+            radii.append(radius)
+
+            obsDate = visSum[0].getVisitInfo().getDate()
+            obsMJD = obsDate.get(obsDate.MJD)
+            mjds.append(obsMJD)
+
+        # Find groups of visits where any one of the visits overlaps another by
+        # a given fraction of the field-of-view radius.
+        distance = self.config.visitOverlap * np.median(radii)
+        clustering = AgglomerativeClustering(
+            distance_threshold=distance.asDegrees(), n_clusters=None, linkage="single"
+        )
+        clusters = clustering.fit(np.array(radecs))
+
+        medianMJD = np.median(mjds)
+        medianEpoch = astropy.time.Time(medianMJD, format="mjd").decimalyear
+
+        fieldNames = []
+        fieldRAs = []
+        fieldDecs = []
+        epochs = []
+        fieldRegions = {}
+
+        for i in range(clusters.n_clusters_):
+            fieldInd = clusters.labels_ == i
+            fieldDetectors = sum([allDetectorCorners[f] for f, fInd in enumerate(fieldInd) if fInd], [])
+            hull = lsst.sphgeom.ConvexPolygon.convexHull(fieldDetectors)
+            center = lsst.geom.SpherePoint(hull.getCentroid())
+            ra = center.getRa().asDegrees()
+            dec = center.getDec().asDegrees()
+
+            fieldRegions[i] = hull
+            fieldNames.append(str(i))
+            fieldRAs.append(ra)
+            fieldDecs.append(dec)
+            epochs.append(medianEpoch)
```

Comment:
I don't think we want to do this, because the catalog of object fit positions will be at the field epochs, and we want that to all be consistent. If we end up fitting visits where the resulting fields would have significantly different median epochs, we may have to reevaluate this.